omegaconf/base.py

```diff
@@ -463,5 +484,25 @@ def _re_parent(self) -> None:
                         if isinstance(item, Container):
                             item._re_parent()
 
+    def _invalidate_flags_cache(self) -> None:
+        from .dictconfig import DictConfig
+        from .listconfig import ListConfig
+
+        # invalidate subtree cache only if the cache is initialized in this node.
+        if self.__dict__["_flags_cache"] is not None:
+            self.__dict__["_flags_cache"] = None
+            if isinstance(self, DictConfig):
+                content = self.__dict__["_content"]
+                if isinstance(content, dict):
+                    for _key, value in self.__dict__["_content"].items():
+                        if value is not None:
```

Comment:
I don't think `value` can be `None` (at least adding `assert value is not None` just before the `if` still passes all tests) -- same for lists below